pkg/kubectl/apply/strategy/strategic_visitor.go

```diff
@@ -69,12 +73,13 @@ func (v delegatingStrategy) MergeMap(diff apply.MapElement) (apply.Result, error
 // MergeType delegates visiting a map based on the field patch strategy.
 // Defaults to "merge"
 func (v delegatingStrategy) MergeType(diff apply.TypeElement) (apply.Result, error) {
-	// TODO: Support retainkeys
 	switch diff.GetFieldMergeType() {
 	case apply.MergeStrategy:
 		return v.merge.MergeType(diff)
 	case apply.ReplaceStrategy:
 		return v.replace.MergeType(diff)
+	case apply.RetainKeysStrategy:
```

Comment:
Adding a new strategy implies that you have 3 places to update here (type, map, list). Shouldn't this be a dispatch rather than a "if/switch" statement?

EDIT: I understand that it might not be that easy ...